sympy/matrices/expressions/matexpr.py

```diff
@@ -549,6 +549,15 @@ def applyfunc(self, func):
         from .applyfunc import ElementwiseApplyFunction
         return ElementwiseApplyFunction(func, self)
 
+    def _eval_Eq(self, other):
+        if not isinstance(other, MatrixExpr):
+            return False
+        if self.shape != other.shape:
+            return False
+        if (self - other).is_ZeroMatrix:
+            return True
+        return Eq(self, other, evaluate=False)
```

Comment:
I might be misremembering but isn't `return None` from `_eval_eq` the way to get an unevaluated `Eq`?

Reply:
See the original issue. Normally you would want to return None, but the way the code is written, when it returns None Eq still applies other tests, which fail on matrices (in particular, `lhs - rhs`).